tachyon/c/zk/air/sp1/baby_bear_poseidon2_opened_values.cc

```diff
@@ -40,3 +41,32 @@ void tachyon_sp1_baby_bear_poseidon2_opened_values_serialize(
   base::Buffer buffer(data, *data_len);
   CHECK(buffer.Write(c::base::native_cast(*opened_values)));
 }
+
+void tachyon_sp1_baby_bear_poseidon2_opened_values_allocate_outer(
+    tachyon_sp1_baby_bear_poseidon2_opened_values* opened_values, size_t round,
+    size_t rows, size_t cols) {
+  std::vector<std::vector<std::vector<math::BabyBear4>>>& mat =
+      c::base::native_cast(*opened_values)[round];
+  mat.resize(rows);
+  for (size_t r = 0; r < rows; ++r) {
+    mat[r].resize(cols);
+  }
+}
+
+void tachyon_sp1_baby_bear_poseidon2_opened_values_allocate_inner(
+    tachyon_sp1_baby_bear_poseidon2_opened_values* opened_values, size_t round,
+    size_t r, size_t cols, size_t size) {
+  std::vector<std::vector<std::vector<math::BabyBear4>>>& mat =
+      c::base::native_cast(*opened_values)[round];
+  std::vector<std::vector<math::BabyBear4>>& row = mat[r];
+  for (size_t c = 0; c < cols; ++c) {
+    row[c].resize(size);
+  }
+}
```

Comment:
Shouldn't this be in the next commit?

Reply:
That's also possible, however by separating out the Rust binding changes from the C API changes, I can make it easier for reviewers to focus specifically on the C API modifications.